### PR TITLE
From territories to towns, fix #386

### LINKED ISSUE #386: Improve territories after demo
* [x] deal with duplicates
* [ ] better wording
* [x]  fix typos
* [ ] ~~put data from existing towns on top~~ → next iteration
* [x] better order of datasets
* [x] add licenses and retro-actions(?) to/from wikipedia
* [x] more information within typeahead dropdown
* [x] change email address

### PR DESCRIPTION
**Pull request by davidbgk**

### CODE REVIEW

File: js/territory/typeahead.js [hunk 1]
```diff
@@ -35,7 +36,7 @@ define([
         source: engine,
         display: 'name',
         templates: {
-            header: header({title: i18n._('Territory')}),
+            header: header({title: i18n._('Territories')}),
```
Comment (davidbgk) [inline js/territory/typeahead.js]: Towns?

File: udata/features/territories/__init__.py [hunk 2]
```diff
@@ -14,38 +14,11 @@ def check_for_territory(query):
...
-                    # Warning: it means that we're dropping some territories
-                    # here, e.g.: 62760 vs. 62814 vs. 80756.
-                    geozones = qs(keys__postal__contains=query)
-                    geozones = geozones.order_by('-population', '-area')
-                    return geozones.first()
-            except GeoZone.DoesNotExist:
-                return False
+        # Match both INSEE and postal codes
+        qs = qs(db.Q(code=query) | db.Q(keys__postal__contains=query))
+        return qs.order_by('-population', '-area')
```
Comment (davidbgk) [inline udata/features/territories/__init__.py]: Useless?

File: udata/templates/search.html [hunk 3]
```diff
@@ -52,11 +52,11 @@
             </a>
         </li>
     {% endif %}
-    {% if territory %}
+    {% if territories %}
         <li>
-            <a href="{{ url_for('territories.territory', territory=territory) }}">
-                {{ _('Territory:') }}
-                <span class="badge">{{ territory.name }}</span>
+            <a href="#territories" data-toggle="tab">
```
Comment (davidbgk) [inline udata/templates/search.html]: I think we should only have a tab when there are more than one result.

File: udata/core/search/views.py [hunk 4]
```diff
@@ -37,5 +37,6 @@ def render_search():
         results_labels = ['datasets', 'reuses', 'organizations', 'users']
     results = search.multiquery(*search_queries)
     context = dict(zip(results_labels, results))
-    context['territory'] = check_for_territory(params.get('q'))
+    territories = check_for_territories(params.get('q'))
+    context['territories'] = territories.paginate(1, search.DEFAULT_PAGE_SIZE)
```
Comment (davidbgk) [inline udata/core/search/views.py]: Doesn't work when territories are not activated.

File: js/territory/typeahead.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ define([
     'templates/search/suggestion.hbs',
     'i18n'
 ], function($, API, Bloodhound, header, suggestion, i18n) {
+    const MAX = 2;
```
Comment (davidbgk) [inline js/territory/typeahead.js]: Maybe extend to more than 2 (5?) given that there are more cases than I thought.

Comment (noirbizarre) [inline js/territory/typeahead.js]: 5 is way too much (we limit dataset to 3). So maybe 3 ?

File: udata/templates/search.html (hunk 3 above)
Comment (noirbizarre) [inline udata/templates/search.html]: OK, I will mix the two solution: 1 result, the tab redirects to the territory page, 2 and more, we list results.

File: udata/templates/search.html [hunk 6]
```diff
@@ -52,12 +52,20 @@
...
         </li>
     {% endif %}
-    {% if territory %}
+    {% if territories %}
         <li>
-            <a href="{{ url_for('territories.territory', territory=territory) }}">
-                {{ _('Territory:') }}
-                <span class="badge">{{ territory.name }}</span>
-            </a>
+            {% set total=territories.count() %}
```
Comment (davidbgk) [inline udata/templates/search.html]: Maybe we should extend that logic to other tabs?

**Pull request merged**